src/map/script.c

```diff
@@ -18722,6 +18728,77 @@ BUILDIN_FUNC(waitingroom2bg_single)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+
+/// Creates an instance of battleground battle group.
+/// *bg_create("<map name>",<x>,<y>{,"<On Quit Event>","<On Death Event>"});
+/// @author [secretdataz]
+BUILDIN_FUNC(bg_create) {
+	const char *map_name, *ev = "", *dev = "";
+	int x, y, mapindex = 0, bg_id;
+
+	map_name = script_getstr(st, 2);
+	if (strcmp(map_name, "-") != 0)
+	{
+		mapindex = mapindex_name2id(map_name);
+		if (mapindex == 0)
+		{ // Invalid Map
+			script_pushint(st, 0);
+			return SCRIPT_CMD_SUCCESS;
+		}
+	}
+
+	x = script_getnum(st, 3);
+	y = script_getnum(st, 4);
+	if(script_hasdata(st, 5))
+		ev = script_getstr(st, 5); // Logout Event
+	if(script_hasdata(st, 6))
+		dev = script_getstr(st, 6); // Die Event
+
+	if (ev[0] != '\0')
+		check_event(st, ev);
+	if (dev[0] != '\0')
+		check_event(st, dev);
+
+	if ((bg_id = bg_create(mapindex, x, y, ev, dev)) == 0)
+	{ // Creation failed
+		script_pushint(st, 0);
```

Comment:
if case is useless, since you just push the same value as when you would return bg_id directly